components/kserve/kserve.go

```diff
@@ -50,8 +50,8 @@ type Kserve struct {
 	// Mesh (Istio) is prerequisite, since it is used as networking layer.
 	Serving infrav1.ServingSpec `json:"serving,omitempty"`
 	// Configures the default deployment mode for Kserve. This can be set to 'Serverless' or 'RawDeployment'.
-	// The value specified in this field will be used to set the default deployment mode in the 'inferenceservice-config' configmap for Kserve
-	// If no default deployment mode is specified, Kserve will use Serverless mode
+	// The value specified in this field will be used to set the default deployment mode in the 'inferenceservice-config' configmap for Kserve.
+	// This field is optional. If no default deployment mode is specified, Kserve will use Serverless mode.
```

Comment:
i will prefer to leave this one to serving team.
to update the comments is only to fix the strange API description. but i can remove this "This field is optional" and leave it for another PR to deal with `DefaultDeploymentMode`

Reply:
The `DefaultDeploymentMode` is just an alias to `string`. Given the `omitempty`, I think it is truly optional unless it is not possible to assign the empty string.

See a screenshot of the resource here: https://github.com/opendatahub-io/opendatahub-operator/pull/864#discussion_r1503288913

I don't recall seeing the field on my own testings... and I do remember trying it myself. So, despite Vedant's screenshot does not give enough additional lines for context, that screenshot should be a complete definition of the kserve component. Notice the `defaultDeploymentMode` field is missing.

Reply:
The screenshot is truncated so I am not sure that the whole picture, but I assumed it's  a struct though now sure why as the name of the type suggests it might just be alias to the string. Sorry for the noise :)